### PR TITLE
Enable LLaVa-1.5 in VLM Pipeline

### PR DESCRIPTION
**Pull request by yatarkan**
Ticket: CVS-153333

### CODE REVIEW

File: src/cpp/include/openvino/genai/vision_encoder.hpp [hunk 1]
```diff
@@ -41,11 +41,16 @@ struct EncodedImage {
 /// ov::InferRequest and configured by ProcessorConfig.
 class OPENVINO_GENAI_EXPORTS VisionEncoder {
 public:
+    /// @brief A string denoting model type.
+    std::string model_type;
```
Comment (Wovchena) [inline src/cpp/include/openvino/genai/vision_encoder.hpp:45]: Can this be a `class enum`? That way we additionally describe the list of supported models.

**Comment by Wovchena** [comment]
Please, resolve the conflict

File: src/cpp/include/openvino/genai/vision_encoder.hpp [hunk 2]
```diff
@@ -41,11 +41,16 @@ struct EncodedImage {
...
     /// @brief A model for image encoding.
     ov::InferRequest m_encoder;
     /// @brief A config to follow.
     ProcessorConfig m_processor_config;
 
+    // LLaVa specific members
+    ov::InferRequest m_vision_embeddings;
```
Comment (ilya-lavrenov) [inline src/cpp/include/openvino/genai/vision_encoder.hpp:52]: should we define base class for VisionEncoder ? and create inherited classes for MiniCPM and LLava specifically 

In this case, we don't need to put all the fields for different models into common class. Similarly, processor config can be model specific class.

File: src/cpp/src/vlm_pipeline.cpp [hunk 3]
```diff
@@ -586,3 +545,138 @@ GenerationConfig VLMPipeline::get_generation_config() const {
...
+        
+        ov::Tensor text_embeds = process_prompt(m_embedding, input_ids, m_vlm_config.scale_emb);
+
+        int64_t image_token_index = 32000; // TODO Consider getting from m_vlm_config.image_token_index or config.json
+
+        return merge_text_and_image_embeddings_llava(input_ids, text_embeds, image_embeds, image_token_index);
+    }
+}
+
+ov::Tensor VLMPipeline::get_inputs_embeds_minicpm(const std::string& prompt, const std::vector<ov::Tensor>& images) {
```
Comment (ilya-lavrenov) [inline src/cpp/src/vlm_pipeline.cpp:568]: can we create a dedicated class responsible for Inputs embedding? 
Such class will be used in Continuous batching implementation 

In this case, VLM Pipeline will compute input embeddings w/o knowing any model / pipeline specific and then perform inference of LLM part in auto-regressive mode

File: src/cpp/include/openvino/genai/vision_encoder.hpp (hunk 1 above)
Comment (yatarkan) [reply src/cpp/include/openvino/genai/vision_encoder.hpp]: Added enum class for model type

**Review by Wovchena — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]
Agreed to perform splitting of implementations of VisionEncoder to model backends and introducing VisionTextInputsEmbedder as separate tasks after llava next / intern vl enablement in GenAI

**Comment by Wovchena** [comment]
There's a conflict again

File: samples/generation.gif
Comment (ilya-lavrenov) [inline samples/generation.gif:1]: please, return this file

**Comment by ilya-lavrenov** [comment]
Please, update a list of supported models https://github.com/openvinotoolkit/openvino.genai/blob/master/src/docs/SUPPORTED_MODELS.md#visual-language-models

File: samples/generation.gif
Comment (yatarkan) [reply samples/generation.gif:1]: Fixed

File: src/docs/SUPPORTED_MODELS.md [hunk 4]
```diff
@@ -167,14 +167,23 @@ The pipeline can work with other similar topologies produced by `optimum-intel`
...
       <td>MiniCPMV</td>
+      <td><code>MiniCPM-V-2_6</code></td>
       <td>
         <ul>
           <li><a href="https://huggingface.co/openbmb/MiniCPM-V-2_6"><code>openbmb/MiniCPM-V-2_6</code></a></li>
         </ul>
       </td>
     </tr>
+    <tr>
+      <td>LLaVA</td>
```
Comment (Wovchena) [inline src/docs/SUPPORTED_MODELS.md:179]: Please, sort it alphabetically by placing above MiniCPM

File: src/cpp/include/openvino/genai/vision_encoder.hpp [hunk 5]
```diff
@@ -41,11 +42,16 @@ struct EncodedImage {
...
 public:
+    /// @brief A enum denoting model type.
+    VLMModelType model_type;
     /// @brief A model for image encoding.
     ov::InferRequest m_encoder;
     /// @brief A config to follow.
     ProcessorConfig m_processor_config;
 
+    // LLaVa specific members
+    ov::InferRequest m_vision_embeddings;
```
Comment (Wovchena) [inline src/cpp/include/openvino/genai/vision_encoder.hpp:53]: Why isn't `m_encoder` reused? You can rename it if you feel `m_vision_embeddings` is a better name.

**Review by Wovchena — CHANGES_REQUESTED** [review]

File: src/docs/SUPPORTED_MODELS.md (hunk 4 above)
Comment (yatarkan) [reply src/docs/SUPPORTED_MODELS.md:179]: Fixed

File: src/cpp/include/openvino/genai/vision_encoder.hpp (hunk 5 above)
Comment (yatarkan) [reply src/cpp/include/openvino/genai/vision_encoder.hpp]: Fixed

**Review by Wovchena — APPROVED** [review]

**Pull request merged**